include/boost/compute/detail/vector_type_producer.hpp

```diff
@@ -0,0 +1,74 @@
+//---------------------------------------------------------------------------//
+// Copyright (c) 2016 Jakub Szuppe <j.szuppe@gmail.com>
+//
+// Distributed under the Boost Software License, Version 1.0
+// See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt
+//
+// See http://boostorg.github.com/compute for more information.
+//---------------------------------------------------------------------------//
+
+#ifndef BOOST_COMPUTE_DETAIL_VECTOR_TYPE_PRODUCER_HPP
+#define BOOST_COMPUTE_DETAIL_VECTOR_TYPE_PRODUCER_HPP
+
+#include <boost/preprocessor/cat.hpp>
+#include <boost/compute/types/fundamental.hpp>
+
+namespace boost {
+namespace compute {
+namespace detail {
+
+/// \internal_
+/// Meta-function returning the vector type based on the scalar type and
+/// the size.
+///
+/// This is useful because ::boost::compute::vector_type<int_, 2> is not
+/// the same as int2_.
+///
+/// For example,
+/// \code
+/// vector_type_producer<float_, 4>::type == float4_
+/// vector_type_producer<int_, 1>::type == int_
+/// \endcode
+template<class Scalar, size_t N>
+struct vector_type_producer
```

Comment:
Isn't this the same as `make_vector_type<T, N>` from `<boost/compute/type_traits/make_vector_type.hpp>`?